api/base/parsers.py

```diff
@@ -15,30 +18,62 @@ class JSONAPIParser(JSONParser):
     media_type = 'application/vnd.api+json'
     renderer_class = JSONAPIRenderer
 
+    @staticmethod
+    def flatten_data(resource_object, request_method, is_list=False):
+        """
+        Flattens data objects, making attributes fields the same level as id and type.
+        """
+
+        if "attributes" not in resource_object and request_method != 'DELETE':
+            raise JSONAPIException(source={'pointer': '/data/attributes'}, detail=NO_ATTRIBUTES_ERROR)
+
+        object_id = resource_object.get('id')
+        object_type = resource_object.get('type')
+
+        # For validating type and id for bulk delete:
+        if is_list and request_method == 'DELETE':
+            if object_id is None:
+                raise JSONAPIException(source={'pointer': '/data/id'}, detail=NO_ID_ERROR)
+
+            if object_type is None:
+                raise JSONAPIException(source={'pointer': '/data/type'}, detail=NO_TYPE_ERROR)
+
+        attributes = resource_object.get('attributes')
+        parsed = {'id': object_id, 'type': object_type}
+        if attributes:
+            parsed.update(attributes)
```

Comment:
Validation in L27 ensures that 'attributes' is present unless the request method is "DELETE".  so later when I try to access attributes, it will be there, unless the request method is DELETE.  That's why I still want to use .get('attributes')